### ISSUE TITLE
Enhancement: Add `max_completion_tokens` for o1 series models

### ISSUE THREAD

**Issue by fitchmultz** [danny-avila/LibreChat#4349]
### What happened?

When using the OpenAI API with the `o1-preview` and `o1-mini` models, an error occurs due to the use of an unsupported parameter. Specifically, the `max_tokens` parameter is not compatible with these models, and `max_completion_tokens` should be used instead.

### Steps to Reproduce

1. Configure the client to use either the `o1-preview` or `o1-mini` model.
2. Set a value for "Max Output Tokens" in the parameters or preset panel of your application.
3. Attempt to make an API call to OpenAI for chat completion.

### Relevant log output

```shell
[OpenAIClient.chatCompletion] Unhandled error type Error: 400 Unsupported parameter: 'max_tokens' is not supported with this model. Use 'max_completion_tokens' instead.
[handleAbortError] AI response error; aborting request: Error: 400 Unsupported parameter: 'max_tokens' is not supported with this model. Use 'max_completion_tokens' instead.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Solution: don’t set a value for max output tokens, since it’s not supported for o1

**Comment by fitchmultz** [comment]
That's not a solution. `max_completion_tokens` has replaced `max_tokens`, which is deprecated. For any non-o1 models they are interchangeable.

Many parameters cannot be changed for o1-preview and o1-mini but response length can be limited by setting `max_completion_tokens`.

Info here: https://platform.openai.com/docs/api-reference/chat/create

![Screenshot_20241007_172128_Chrome.png](https://github.com/user-attachments/assets/ae83e486-701b-499a-9d93-0863a9bcf3e5)

**Comment by danny-avila** [comment]
I see, thank you for clarifying I was unaware of the distinction. I will replace `max_tokens` with `max_completion_tokens` for o1 models only for maximum backwards compatibility, and will keep it in mind that `max_tokens` is technically deprecated.

**Comment by danny-avila** [comment]
Closed by #4376
